### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

**Review by Alexhqolma — APPROVED** [review]

File: src/main/java/taxi/controller/driver/AddDriverController.java [hunk 1]
```diff
@@ -24,7 +24,9 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
     public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         String name = req.getParameter("name");
         String licenseNumber = req.getParameter("license_number");
-        Driver driver = new Driver(name, licenseNumber);
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        Driver driver = new Driver(name, licenseNumber, login, password);
```
Comment (Noct2000) [inline src/main/java/taxi/controller/driver/AddDriverController.java:29]: we have RegistrationController, do we need this controller and vice versa?

File: src/main/java/taxi/controller/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession session = req.getSession();
+
+        Long driverId = (Long) session.getAttribute("driver_id");
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null) {
+            resp.sendRedirect("/login");
```
Comment (Noct2000) [inline src/main/java/taxi/controller/web/filter/AuthenticationFilter.java:40]: add context path before all URLs in the `resp.sendRedirect`

File: src/main/webapp/WEB-INF/views/authentication/register.jsp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        Name <input type="text" name="name" required><br>
+        License number<input type="text" name="licenseNumber" required><br>
+        Login <input type="text" name="login" required><br>
+        Password <input type="password" name="password" required><br>
+        Repeat password <input type="password" name="repeatPassword" required><br>
+        <button type="submit">Register</button>
+    </form>
+  <h4><a href="${pageContext.request.contextPath}/login">Login</a></h4>
+</body>
+</html>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/authentication/register.jsp:19]: add new line to the end of the file

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/controller/driver/AddDriverController.java (hunk 1 above)
Comment (AndrewHudzenko) [reply src/main/java/taxi/controller/driver/AddDriverController.java:29]: what do you mean, which controller ? RegistrationController or AddDriverController? if RegistrationController -yes, we need it, if AddDriverController -yes, we need it too

File: src/main/java/taxi/controller/authentication/LoginController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpSession;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Injector;
+import taxi.model.Driver;
+import taxi.service.AuthenticationService;
+
+public class LoginController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private static final AuthenticationService authenticationService
+            = (AuthenticationService) injector.getInstance(AuthenticationService.class);
```
Comment (Noct2000) [inline src/main/java/taxi/controller/authentication/LoginController.java:17]: don't move `=` to the new line, check your code

File: src/main/webapp/WEB-INF/views/drivers/add.jsp [hunk 5]
```diff
@@ -5,28 +5,37 @@
...
-</table>
+<%@include file='/WEB-INF/views/header.jsp'%>
+    <form method="post" id="driver" action="${pageContext.request.contextPath}/drivers/add"></form>
+        <h1 class="table_dark">Add driver:</h1>
+        <table border="1" class="table_dark">
+            <tr>
+                <th>Name</th>
+                <th>License number</th>
+                <th>Login</th>
+                <th>Password</th>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/drivers/add.jsp:19]: password output is bad practice, let's remove it

File: src/main/webapp/WEB-INF/views/drivers/cars.jsp [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        </c:forEach>
+      </td>
+      <td>
+        <a href="${pageContext.request.contextPath}/cars/delete?id=${car.id}">DELETE</a>
+      </td>
+    </tr>
+  </c:forEach>
+</table>
+</body>
+</html>
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/drivers/cars.jsp:47]: add new line to the end of the file

File: src/main/java/taxi/service/RegisterService.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package taxi.service;
+
+import taxi.exception.AuthenticationException;
+import taxi.model.Driver;
+
+public interface RegisterService {
+    Driver register(String name, String licenseNumber,
+                    String login, String password, String repeatPassword)
+            throws AuthenticationException;
+}
```
Comment (Noct2000) [inline src/main/java/taxi/service/RegisterService.java:10]: I can't find a implementation of this service

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/views/drivers/add.jsp (hunk 5 above)
Comment (AndrewHudzenko) [reply src/main/webapp/WEB-INF/views/drivers/add.jsp:19]: yes, but it is not up to me, the condition says that our driver will be as user, so how can we create a new driver without this data?

File: src/main/java/taxi/controller/car/GetMyCurrentCarsController.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import taxi.lib.Injector;
+import taxi.model.Car;
+import taxi.service.CarService;
+
+public class GetMyCurrentCarsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private static final CarService carService
+            = (CarService) injector.getInstance(CarService.class);
```
Comment (Noct2000) [inline src/main/java/taxi/controller/car/GetMyCurrentCarsController.java:17]: don't move `=` to the new line, check your code

File: src/main/java/taxi/controller/web/filter/AuthenticationFilter.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/login");
+        allowedUrls.add("/register");
+        allowedUrls.add("/index");
```
Comment (Noct2000) [inline src/main/java/taxi/controller/web/filter/AuthenticationFilter.java:23]: this is redundant 
image
```suggestion
```

File: src/main/webapp/WEB-INF/views/css/table_dark.css [hunk 10]
```diff
@@ -30,3 +38,12 @@
...
 }
+
+#logoutId {
+    align-content: center;
+    background-color: silver;
+    border-radius: 4px;
+    text-align: center;
+    border: 1px solid #98E2CA;
+    height: 23px;
+}
```
Comment (Noct2000) [inline src/main/webapp/WEB-INF/views/css/table_dark.css:49]: add new line to the end of the file

**Review by Noct2000 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/controller/web/filter/AuthenticationFilter.java (hunk 9 above)
Comment (AndrewHudzenko) [reply src/main/java/taxi/controller/web/filter/AuthenticationFilter.java:23]: yes, i just forget to remove this/ we were testing one case with Sophia with this

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**